### PR TITLE
Community Token Holder

### PR DESCRIPTION
**Pull request by Janther**
also some updates to the compiler versions

### CODE REVIEW

File: contracts/CommunityTokenHolder.sol [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    uint256 canExtract = total.mul(extractableFraction()).div(29);
+
+    canExtract = canExtract.sub(collectedTokens);
+
+    if (canExtract > balance) {
+      canExtract = balance;
+    }
+
+    collectedTokens = collectedTokens.add(canExtract);
+    assert(aix.transfer(controller, canExtract));
```
Comment (rstormsf) [inline contracts/CommunityTokenHolder.sol]: still using `assert`

File: contracts/CommunityTokenHolder.sol [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    contribution = Contribution(_contribution);
+    aix = ERC20(_aix);
+  }
+
+  /// @notice The Dev (Owner) will call this method to extract the tokens
+  function collectTokens() public onlyController {
+    uint256 balance = aix.balanceOf(address(this));
+    uint256 total = collectedTokens.add(balance);
+    // This wallet will get a 29% of the total tokens.
+    // since scaling 4 of 29 tto a percentage looks horrible (13.7931034483),
```
Comment (matt-lough) [inline contracts/CommunityTokenHolder.sol]: :s/tto/to

**Review by rstormsf — APPROVED** [review]

File: contracts/CommunityTokenHolder.sol (hunk 2 above)
Comment (rstormsf) [reply contracts/CommunityTokenHolder.sol]: tto :-)

File: contracts/CommunityTokenHolder.sol (hunk 1 above)
Comment (Janther) [reply contracts/CommunityTokenHolder.sol]: 👍 forgot about this

File: contracts/CommunityTokenHolder.sol (hunk 2 above)
Comment (IsaacPayne) [inline contracts/CommunityTokenHolder.sol]: Typo should be `to` rather than `tto`

**Review by matt-lough — APPROVED** [review]

File: contracts/CommunityTokenHolder.sol [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    TokensWithdrawn(controller, canExtract);
+  }
+
+  function extractableFraction() constant returns (uint256) {
+    uint256 finalizedTime = contribution.finalizedTime();
+    require(finalizedTime > 0);
+
+    uint256 timePassed = getTime().sub(finalizedTime);
+
+    if (timePassed > months(12)) {
```
Comment (IsaacPayne) [inline contracts/CommunityTokenHolder.sol:47]: Maybe a little comment about what is going on here?

**Review by IsaacPayne — APPROVED** [review]

**Pull request merged**